developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,421 @@
+  (user decide how to use version parameter);
+- *version_minor* - shall be used only for internal changes (optimizations,
+  bug fix);
+- *version_hotfix* - shall be used during the bug fixing period;
+- *version_build* - shall be incremented with new build.
+
+Module affinity
+---------------
+**Usage:**
+``affinity_mask mask``
+
+**Example:**
+``affinity_mask MASTER_CORE_AFFINITY``
+
+*mask* is a bit-mask of cores allowed to execute module. Macros reffering this
+mask are defined at:
+``<repo_dir>/FW/portable/include/modules.h``
+
+Module instance count
+---------------------
+**Usage:**
+``instance_count count``
+
+**Example:**
+``instance_count +1``
+
+*instance_count* refers to number of module instance. The value should be
+provided with "+" at beginning. It will be added to basic instance count.
+
+Scheduler domain type
+---------------------
+**Usage:**
+``domain_types type``
+
+**Example:**
+``domain_types DP``
+
+Following domain types are available :
+- *LL* - low latency domain is intended to be able transfer data through DSP
+subsystem within 2ms. Processing modules used in that domain must be able to
+work on limited number of samples. To meet 2ms pipeline latency on stream
+sampled at 48 kHz stream needs to be processed in 1ms periods i.e. using maximum
+48 sample blocks.
+- *DP* - Data processing domain is intended host all processing that does not
+fit in definition of low latency domain. It is important that this domain is
+still real time and latency sensitive.
+
+Module type
+-----------
+**Usage:**
+``type class_name``
+
+**Example:**
+``type AudClassModule``
+
+*class_name* refers to module class type supported by firmware. Available
+modules classes are placed in:
+``<repo_dir>/FW/intel_common/module_binmaps/dsp_fw_common.binmap``
+in ``# type <mod_func_type>`` section.
+
+.. code-block:: bash
+
+  enum module_type {
+        basefw          = 0,
+        mixin           = 1,
+        mixout          = 2,
+        copier          = 3,
+        peakvol         = 4,
+        updwmix         = 5,
+        mux             = 6,
+        src             = 7,
+        wov             = 8,
+        fx              = 9,
+        aec             = 10,
+        kpb             = 11,
+        micselect       = 12,
+        fxf             = 13,
+        audclass        = 14,
+        fakecopier      = 15,
+        iodriver        = 16,
+        whm             = 17,
+        gdbstub         = 18,
+        sensing         = 19,
+        max             = 20,
+        invalid         = emax
+   };
+
+auto_start
+----------
+**Usage:**
+``auto_start true or false`` Indicates whether a instance of the module should be created at the base fw startup
+
+**Example:**
+``auto_start false``
+
+init_config
+-----------
+**Usage:**
+``init_config true or false`` 0 (only basic config in payload) or 1 (basic config with extension that contains pin format).
+
+**Example:**
+``init_config 1``
```

Comment:
can this be `init_config_has_extension` just so its clear what the bool is identifing?

Reply:
yes i think a rename is warranted here

Reply:
yes, agree